nixos/lib/utils.nix

```diff
@@ -14,8 +14,20 @@ rec {
   fsNeededForBoot = fs: fs.neededForBoot || elem fs.mountPoint pathsNeededForBoot;
 
   # Check whenever `b` depends on `a` as a fileSystem
-  fsBefore = a: b: a.mountPoint == b.device
-                || hasPrefix "${a.mountPoint}${optionalString (!(hasSuffix "/" a.mountPoint)) "/"}" b.mountPoint;
+  fsBefore = a: b:
+    let
+      normalisePath = path: "${path}${optionalString (!(hasSuffix "/" path)) "/"}";
+      normalise = mount: mount // { device = normalisePath mount.device;
+                                    mountPoint = normalisePath mount.mountPoint;
+                                    depends = map normalisePath mount.depends;
+                                  };
+
+      a' = normalise a;
+      b' = normalise b;
+
+    in hasPrefix a'.mountPoint b'.device
+    || hasPrefix a'.mountPoint b'.mountPoint
+    || any (dependency: hasPrefix a'.mountPoint dependency) b'.depends;
```

Comment:
There is definitely no reason to normalise `a` because `a` "is a prefix of" `a'`.
I really think normalising doesn't make it more correct, just harder to read. But it would be nice ti hear other people's opinions.

Reply:
That would give false positives: `/aaa` is not a parent of `/aaaa`, but it would be handled as one without `normalise`

Reply:
Now I get it. Thanks!